doc/reference/kernel/threads/nothread.rst

```diff
@@ -0,0 +1,73 @@
+.. _nothread:
+
+Zephyr Without Threads
+######################
+
+Thread support is not necessary in some applications:
+
+* Bootloaders
+* Simple event-driven applications
+* Examples intended to demonstrate core functionality
+
+Thread support can be disabled in Zephyr by setting
+:option:`CONFIG_MULTITHREADING` to ``n``.  Since this configuration has
+a significant impact on Zephyr's functionality and testing of it has
+been limited, there are conditions on what can be expected to work in
+this configuration:
+
+* The :ref:`build system <application>`
+
+* The ability to boot the application to ``main()``
+
+* :ref:`Interrupt management <interrupts_v2>`
+
+* Non-sleeping delays e.g. :c:func:`k_busy_wait`
+
+* The system clock including :c:func:`k_uptime_get`
+
+* Specifically identified drivers in certain subsystems, listed below.
+
+Some existing drivers do not work when threading is disabled, but are
+within scope based on their subsystem, or may be sufficiently isolated
+that supporting them on a particular platform is low-impact.
+Enhancements to add support to existing capabilities that were not
+originally implemented to work with threads disabled will be considered.
+
+.. contents::
+    :local:
+    :depth: 1
+
+Flash
+=====
+
+The :ref:`flash_api` is expected to work for all SoC flash peripheral
+drivers.  Bus-accessed devices like serial memories may not be
+supported.
+
+*List/table of supported drivers to go here*
+
+GPIO
+====
+
+The :ref:`gpio_api` is expected to work for all SoC GPIO peripheral
+drivers.  Bus-accessed devices like GPIO extenders may not be supported.
+
+*List/table of supported drivers to go here*
+
+UART
+====
+
+A subset of the :ref:`uart_api` is expected to work for all SoC UART
+peripheral drivers.
+
+* Applications that select :option:`CONFIG_UART_INTERRUPT_DRIVEN` may
+  work, depending on driver implementation.
+
+* Applications that select :option:`CONFIG_UART_ASYNC_API` may
```

Comment:
I am not sure about this one. I always thought this actually required at least `k_timer`:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/uart_nrfx_uarte.c#L830

Reply:
That's unfortunate.  I wouldn't expect other SoC's to have that problem.  I suppose we don't support that API, then.